pkg/metrics/metrics.go

```diff
@@ -167,6 +172,15 @@ var (
 		[]string{NAMESPACE_LABEL, NAME_LABEL, VERSION_LABEL, PHASE_LABEL, REASON_LABEL},
 	)
 
+	dependencyResolutionSummary = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "olm_resolution_duration_seconds",
+			Help:       "The duration for a dependency to get resolved",
```

Comment:
This technically resolves all dependencies, right?

Reply:
@awgreene it records all dependencies, successful and failed ones.

Reply:
Yes, you could say that this metric is something like "the duration of a dependency resolution attempt". One attempt considers everything that is relevant (i.e. subscriptions, catalogs, CSVs) in a set of namespaces.

Reply:
I like @benluddy 's suggestion - it clearly portrays that OLM spent time calculating all dependencies.